lib/opencensus_phoenix/instrumenter.ex

```diff
@@ -15,7 +15,12 @@ defmodule OpencensusPhoenix.Instrumenter do
 
   @doc false
   def phoenix_controller_call(:start, _compiled_meta, %{conn: conn}) do
-    parent_span_ctx = :oc_propagation_http_tracecontext.from_headers(conn.req_headers)
+    parent_span_ctx = case conn.private[:opencensus_span_ctx] do
```

Comment:
I think I would use `Keyword.has_key?(conn.private, :opencensus_span_ctx)`.

Reply:
If it was only checking it exists that'd make sense, but it actually uses the value as the parent context.

Reply:
Oh yeah, in that case it makes more sense, but maybe use `Keyword.get_lazy/3` instead? That will handle it for you, but still can return `nil` if there is `Plug.Conn.put_private(conn, :opencensus_span_ctx, nil)` somewhere.

Reply:
Not sure I understand. Had to lookup`get_lazy`, it takes a function to evaluate if the key isn't found for cases that value is expensive to calculate?

How would that be used here? And `conn.private[:opencensus_span_ctx]` will return `nil` already if `put_private is passed `nil`, in which case we should get the headers and create the context as it is doing.

Reply:
ooooh, ok, yea I like that